nodeup/pkg/model/ntp.go

```diff
@@ -46,22 +49,102 @@ func (b *NTPBuilder) Build(c *fi.ModelBuilderContext) error {
 		return nil
 	}
 
+	var ntp string
+	switch b.Cluster.Spec.CloudProvider {
+	case "aws":
+		ntp = "169.254.169.123"
+	case "gce":
+		ntp = "time.google.com"
+	default:
+		ntp = ""
+	}
+
 	if b.Distribution.IsDebianFamily() {
 		c.AddTask(&nodetasks.Package{Name: "ntp"})
+
+		if ntp != "" {
+			bytes, err := UpdateNtpIP(ntp, "ntp")
+			if err != nil {
+				return err
+			}
+			c.AddTask(&nodetasks.File{
+				Path:     "/etc/ntp.conf",
+				Contents: fi.NewBytesResource(bytes),
+				Type:     nodetasks.FileType_File,
+				Mode:     s("0644"),
+			})
+		}
+
 		c.AddTask((&nodetasks.Service{Name: "ntp"}).InitDefaults())
 	} else if b.Distribution.IsRHELFamily() {
 		switch b.Distribution {
 		case distros.DistributionCentos8, distros.DistributionRhel8:
 			c.AddTask(&nodetasks.Package{Name: "chrony"})
+
+			if ntp != "" {
+				bytes, err := UpdateNtpIP(ntp, "chrony")
+				if err != nil {
+					return err
+				}
+				c.AddTask(&nodetasks.File{
+					Path:     "/etc/chrony.conf",
+					Contents: fi.NewBytesResource(bytes),
+					Type:     nodetasks.FileType_File,
+					Mode:     s("0644"),
+				})
+			}
 			c.AddTask((&nodetasks.Service{Name: "chronyd"}).InitDefaults())
+
 		default:
 			c.AddTask(&nodetasks.Package{Name: "ntp"})
+
+			if ntp != "" {
+				bytes, err := UpdateNtpIP(ntp, "ntp")
+				if err != nil {
+					return err
+				}
+				c.AddTask(&nodetasks.File{
+					Path:     "/etc/ntp.conf",
+					Contents: fi.NewBytesResource(bytes),
+					Type:     nodetasks.FileType_File,
+					Mode:     s("0644"),
+				})
+			}
+
 			c.AddTask((&nodetasks.Service{Name: "ntpd"}).InitDefaults())
 		}
 	} else {
 		klog.Warningf("unknown distribution, skipping ntp install: %v", b.Distribution)
 		return nil
 	}
-
 	return nil
 }
+
+func UpdateNtpIP(ip string, ntp string) ([]byte, error) {
```

Comment:
Seems to me that `ntp` should be its own type, with constants.

Reply:
Thanks @johngmyers I was planning on cleaning that up a bit once I got it working.  Does the newest commit look better?